drivers/rtc/rtc_ll_stm32.c

```diff
@@ -152,8 +150,6 @@ static int rtc_stm32_init(const struct device *dev)
 
 	LL_RCC_EnableRTC();
```

Comment:
HSEM lock is required but only around `LL_RCC_EnableRTC()`.

So it should be: 
```
z_stm32_hsem_lock(CFG_HW_RCC_SEMID, HSEM_LOCK_DEFAULT_RETRY);

LL_RCC_EnableRTC();

z_stm32_hsem_unlock(CFG_HW_RCC_SEMID);
```

Reply:
Got it. I'm still learning, sorry.